core/modules/book/book.admin.inc

```diff
@@ -50,12 +50,18 @@ function book_admin_overview() {
 function book_admin_settings($form, &$form_state) {
   $types = node_type_get_names();
   $config = config('book.settings');
+  $form['book_allowed_all'] = array(
```

Comment:
Just like the UI, it would be good to clarify this is an admin-only option and not directly in conflict with `book_allowed_types`.

```suggestion
  $form['book_admin_allowed_all'] = array(
```